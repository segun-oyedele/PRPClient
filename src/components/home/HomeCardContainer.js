import PropTypes from 'prop-types';
import { HomeCardItem } from './';

export const HomeCardContainer = ({ items }) => {
  return (

    <div className="flex flex-col content-between min-h-full gap-6 grid-row-3 home__cards-containers xl:flex-row">
      {
        items.map((item) => (
          <HomeCardItem
            key={item.title}
            { ...item }
          />
        ))
      }
    </div>
  );
};

HomeCardContainer.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape(
    { 
      title: PropTypes.string.isRequired, 
      description: PropTypes.string.isRequired 
    }
  )).isRequired
};
