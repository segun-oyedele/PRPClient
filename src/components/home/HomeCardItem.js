import Image from 'next/image';
import PropTypes from 'prop-types';

export const HomeCardItem = ({ title, description, iconName }) => {
  return (
    <div className="flex flex-col justify-between w-full max-w-md gap-10 px-8 py-4 rounded-2xl home_card-report">

      <div>
        <Image
          alt="Card icon"
          className="rounded-lg"
          height={80}
          src={`${process.env.iisPath}/img/icons/${iconName}.svg`}
          title="Card icon"
          width={80}
        />
      </div>
      
      <div className="flex flex-col gap-4">
        <h3 className="text-3xl font-semibold home_card-title">{title}</h3>
        <p className="mt-2 home_card-description">{description}</p>
      </div>

    </div>
  );
};

HomeCardItem.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};
