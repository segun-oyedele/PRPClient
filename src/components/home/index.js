export * from './HomeCardContainer';
export * from './HomeCardItem';