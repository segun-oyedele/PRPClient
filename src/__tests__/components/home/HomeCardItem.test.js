import { render, screen, fireEvent } from '@testing-library/react';
import { HomeCardItem } from '@/components/home/HomeCardItem';

describe('Test on HomeCardItem', () => {
  test('should match with snapshot', () => {
    const { container } = render(<HomeCardItem title="" description="" />);

    expect(container).toMatchSnapshot();
  })
  
  test('should to show the title and description specified', () => {
    const { container } = render(<HomeCardItem title="title" description="description" />);
    
    expect(screen.getByText('title')).toBeInTheDocument();
    expect(screen.getByText('description')).toBeInTheDocument();
  })
})