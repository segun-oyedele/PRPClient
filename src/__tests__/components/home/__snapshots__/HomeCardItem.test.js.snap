// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on HomeCardItem should match with snapshot 1`] = `
<div>
  <div
    class="flex flex-col justify-between max-w-md px-8 py-4 bg-white rounded-lg shadow-lg w-72 h-60"
  >
    <section>
      <h3
        class="text-xl font-semibold text-center text-gray-800"
      />
      <p
        class="mt-2 text-gray-600"
      />
    </section>
    <div
      class="flex justify-end mt-4"
    >
      <a
        class="text-xl font-medium text-indigo-500 transition-opacity duration-300 cursor-pointer hover:opacity-70"
      >
        Go there
      </a>
    </div>
  </div>
</div>
`;
